models/refugee.py

```diff
@@ -0,0 +1,126 @@
+from datetime import date, datetime
+from enum import Enum
+
+class Refugee:
+    """
+    A class to represent a refugee family.
+    """
+    class MedicalCondition(Enum):
+        CANCER = "Cancer"
+        CHRONICKIDNEY = "Chronic kidney disease"
+        CHRONICLIVER = "Chronic liver disease"
+        CHRONICLUNG = "Chronic lung diseases"
+        DEMENTIA_OR_NEUROLOGICAL = "Dementia or other neurological conditions"
+        DIABETES = "Diabetes"
+        DOWNSYNDROME = "Down Syndrome"
+        HEART = "Heart conditions"
+        HIV = "HIV infection"
+        IMMUNOCOMPROMISED = "Immunocompromised state (weakened immune system)"
+        MENTALHEALTH = "Mental health conditions"
+        OVERWEIGHT_OR_OBESITY = "Overweight and obesity"
+        PREGNANCY = "Pregnancy"
+        SICKLECELL_OR_THALASSEMIA = "Sickle cell or thalassemia"
+        SMOKING = "Smoking, current or former"
+        TRANSPLANT = "Solid organ or blood stem cell transplant"
+        STROKE_OR_CEREBROVASCULAR = "Stroke or cerebrovascular disease, which affects blood flow to the brain"
+        SUBSTANCE_USE = "Substance use disorders"
+        TUBERCULOSIS = "Tuberculosis"
+        OTHERS = "Others"
+
+    def __init__(self,
+                 firstname: str,
+                 lastname:str,
+                 camp: str,
+                 numOfFamilyMember: int,
+                 startingDate:date,
+                 medicalConditionType=None):
+
+        """
+        :param firstname: firstname of refugee family
+        :param lastname: lastname of refugee family
+        :param camp: camp of the refugee family locating at
+        :param numOfFamilyMember: param number of family member
+        :param startingDate: start date of refugee family creation
+        :param medicalConditionType: medical condition type of the refugee
+        """
+
+        self.name = self.__sanitise_name(firstname, lastname)
+        self.numOfFamilyMember = self.__sanitise_num_of_family_member(numOfFamilyMember)
+        self.camp = camp
+        self.startingDate = self.__sanitise_starting_date(startingDate)
+        self.medicalConditionType : set[self.MedicalCondition] = set(medicalConditionType)
```

Comment:
Since the default value of `medicalConditionType` is `None`, if the instance is created without a medical condition, the value will become `set(None)`, while what we want is a empty set. 
(e.g. set(None) has len=1, while set() has len=0)
One way to do that is to first check if `medicalConditionType` is `None`, and if so make it an empty set